api/nakadi-event-bus-api.yaml

```diff
@@ -1503,11 +1535,37 @@ definitions:
         type: string
         default: 'random'
 
+      compatibility_mode:
+        description: |
+          Compatibility modes provide a mean for event owners to evolve their schema, given changes
+          respect the semantics defined by this field.
+
+          It's designed to be flexible enough so that producers can evolve their schemas while not
+          inadvertently breaking existent consumers.
+
+          Once defined, the compatibility mode is fixed, since otherwise it would break a predefined contract,
+          jeopardizing the value of this feature.
+
+          List of available compatibility modes:
+
+          - 'deprecated': Schema changes are not allowed. This is used only by legacy event types. This option is not
+            available for new event types. Schema validation is lenient and allow fields that are not defined. Also,
+            it's possible to use the full json schema specification for defining schemas.
+
+          - 'incompatible': Any schema modification is accepted, even if it might break existing producers or consumers.
+
+          - 'compatible': Schema changes never break existing consumers. Every event published since the first version
```

Comment:
In a nutshell, our design diverged from Avro's because Avro compatibility mode only applies to reading events, not writing.

When we decided to allow schemas to be evolved in a non coordinated manner (not notifying producers or consumers), we constrained the problem, reducing the possibilities of having several modes.

The current solution is simpler to understand, train and debug. It works as `all` compatibility addressing all requirements.